staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/monitoring.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conversion
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	namespace = "apiserver"
+	subsystem = "crd"
+)
+
+var (
+	latencyBuckets = prometheus.ExponentialBuckets(1, 2, 5)
```

Comment:
what do these boundaries represent?

Reply:
Actually I chose them arbitrarily. Thinking more about this, maybe we want to start with smaller numbers:

< 50ms Very Good converter
< 100ms Good converter
< 200ms fair
< 400ms ok
< 1s slow
< 5s very slow
others: very very slow

wdyt?

Reply:
As webhooks can wildly vary and it's hard to predict as we don't have control over them, a common solution is to use a summary when interacting with 3rd party providers. I'm not sure aggregating client latencies against webhooks fleet wide is a very strong use case, I'd rather expect the webhook server to expose those metrics as a histogram and do the fleet wide analysis on that side.

Reply:
First we may start providing webhooks ourselves as standard types eventually move to CRD (or new standard types at least). Also I think summary metric can be derived from histogram metric, right?
Depends on number of times we call conversion webhooks, we may want them to be less than some number to make sure the API call is fast enough. I think these buckets can change in future but histogram make more sense. Also we do have histogram metrics for admission webhooks and if we decided we want summary here, we should be consistent.

Reply:
I can't tell if this discussion got resolved or not. given [discussion of stability guarantees around metrics](https://groups.google.com//forum/#!topic/kubernetes-api-reviewers/kdxON6-TtNE), how sure are we that this metric (name and structure) will not change?

Reply:
I think it is generally preferable to use a histogram than summary metrics. In respect to the aggregation issue, while I agree there is probably a marginal usecases for aggregating this across a fleet, it does feel safer to me to leave that door open and just opt to use a histogram. 

Insofar as the bucket sizing, I agree with starting with smaller numbers. It seems like we opt to use [this bucket sizing](https://github.com/kubernetes/kubernetes/blob/9b8c58644a51fb79386c11bf6b4a31cbbad8417e/pkg/scheduler/metrics/metrics.go#L91) for latency measurements (but obviously this should be shaped to the thing being measured).

Reply:
Changed the bucket sizing to what @logicalhan suggested

Reply:
that bucketing sounds good to me